clang/lib/Sema/SemaExpr.cpp

```diff
@@ -21111,11 +21111,15 @@ ExprResult Sema::CheckPlaceholderExpr(Expr *E) {
     const bool IsTypeAliasTemplateDecl = isa<TypeAliasTemplateDecl>(Temp);
 
     NestedNameSpecifier *NNS = ULE->getQualifierLoc().getNestedNameSpecifier();
-    TemplateName TN(dyn_cast<TemplateDecl>(Temp));
-    if (TN.isNull())
+    // FIXME: AssumedTemplate is not very appropriate for error recovery here,
```

Comment:
Its a shame we cannot do a `getQualifiedAssumedTemplateName` here, and only treat the base-name as `assumed`.

Reply:
There are other solutions to explore.

It seems that for this specific error, we could fish out the actual template declaration this is referring to from the NestedNameSpecifier. Or maybe there is something we could do with the name lookup implementation to improve this and return an actual template.

But this would be a bunch of work for an error recovery path that stayed untested up until this point.